routes/validate.js

```diff
@@ -1,18 +1,21 @@
+var path = require("path");
 module.exports = function(req, res, next) {
-	var token = req.url.split("/");
-	if(token[3]){
-		req.users.findOne({
-			_id: token[3]
-		},function(err, res1) {
-			if (err) {
-				res.json('you are not authenticated');
-			} else if(!res1){
-				res.json('you are not authenticated');
-			}else {
-				next();
-			}
-		})
-	}else{
-		next();
-	}
+    var token = req.url.split("/");
+    console.log(token[3])
+    if (req.method != "POST") {
+        req.access_token.findOne({
+            userid: token[3]
```

Comment:
here you are checking if user id exists in access_token table. you are not checking here if access token is valid or not??
what is the logic of this??